psutil/_psutil_bsd.c

```diff
@@ -391,6 +396,104 @@ psutil_proc_cmdline(PyObject *self, PyObject *args) {
 }
 
 
+/*
+ * Return process environment as a Python dictionary
+ */
+PyObject *
+psutil_proc_environ(PyObject *self, PyObject *args) {
+    unsigned int i, cnt = 0;
+    long pid;
+    char **envs;
+    char errbuf[_POSIX2_LINE_MAX];
+    kvm_t *kd;
+#ifdef PSUTIL_NETBSD
+    struct kinfo_proc2 *p;
+#else
+    struct kinfo_proc *p;
+#endif
+    PyObject *py_retdict;
+
+    if (!PyArg_ParseTuple(args, "l", &pid))
+        return NULL;
+
+#if defined(PSUTIL_FREEBSD)
+    kd = kvm_openfiles(NULL, "/dev/null", NULL, 0, errbuf);
+#else
+    kd = kvm_openfiles(NULL, NULL, NULL, KVM_NO_FILES, errbuf);
+#endif
+    if (!kd) {
+        PyErr_SetString(PyExc_RuntimeError, errbuf);
```

Comment:
There is a utility function, currently exposed for OpenBSD only, which can convert this kvm error:
https://github.com/giampaolo/psutil/blob/994c429c010049a6a0556ce3b0d1af1f86f27867/psutil/arch/openbsd/specific.c#L50-L61

I think it makes sense to move `convert_kvm_err` out of `psutil/arch/openbsd/specific.c` and put it into `psutil/_psutil_common.c` (guarded by an `#ifdef BSD...`) so that you can invoke it from this file, like this:

```c
    if (! kd) {
        convert_kvm_err("kvm_openfiles", errbuf);
        return NULL;
    }
```

Reply:
Ah ok, nice, I've moved it.